src/smartplaylists/searchtermwidget.cpp

```diff
@@ -163,7 +163,11 @@ void SearchTermWidget::FieldChanged(int index) {
       page = ui_->page_rating;
       break;
     case SearchTerm::Type_Text:
-      page = ui_->page_text;
+      if (ui_->op->currentIndex() == 4 || ui_->op->currentIndex() == 5) {
```

Comment:
What are these magic numbers about?

Reply:
Same as in `SearchTermWidget::OpChanged`. The application uses the index of the entry in the combobox. I didn’t find a way how I would be able to do this check otherwise.

Reply:
I just mirrored what the original code does here: https://github.com/kleinfreund/Clementine/blob/4ab6f359d50bb23c116b6dca2a8fc02f26f5cd7a/src/smartplaylists/searchtermwidget.cpp#L214-L216